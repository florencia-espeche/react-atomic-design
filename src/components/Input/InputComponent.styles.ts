import styled, { css } from 'styled-components'


interface TopContainerStyle {

}
interface InputComponentStylesProps {
    isActive?: boolean
    variant?: string
    isEyeOn?: boolean
    error?: boolean
    isFilled?: boolean
    disabled?: boolean
    inputSize?: 'large' | 'medium'
    label?: string
}

export const AlertMessage = styled.div`
    display: flex;
    align-items: center;
    justify-content: flex-start;
    padding: 5px;
    color: ${props => props.theme.colors.suport.error};
    svg {
        margin-right: 5px;
    }
`

export const Container = styled.div<InputComponentStylesProps>`
    box-sizing: border-box;
    position: relative;
    width: auto;
    ${props =>
        props.inputSize === 'large' &&
        css`
            height: 48px;
        `}
    ${props =>
        props.inputSize === 'medium' &&
        css`
            height: 32px;
        `}
    ${props =>
        props.disabled &&
        css`
            cursor: not-allowed;
        `}
`

export const Svg = styled.div`
    display: flex;
    svg {
        right: -2.9rem;
        top: 0.9rem;
        cursor: pointer;
    }
`
export const Mediumlabel = styled.div<InputComponentStylesProps>`
    color: ${props => props.theme.colors.neutral.dark};
    cursor: text;
    ${props => props.theme.fontTokens[75]}
    ${props =>
        props.isActive &&
        css`
            background-color: ${props => props.theme.colors.neutral.white};
            color: ${props => props.theme.colors.primary};
        `}
        ${props =>
        props.disabled &&
        css`
            cursor: not-allowed;
        `}
    ${props =>
        props.error &&
        css`
            color: ${props => props.theme.colors.suport.error};
        `}
`

export const Label = styled.label<InputComponentStylesProps>`
    position: absolute;
    left: 1rem;
    top: 35%;
    padding: 0 0.5rem;
    margin-left: -0.5rem;
    color: ${props => props.theme.colors.neutral.soft};
    cursor: text;
    transition: top 200ms ease-in, left 200ms ease-in, font-size 200ms ease-in;

    ${props =>
        props.isActive &&
        css`
            top: -0.5rem;
            font-size: 0.8rem;
            left: 0.8rem;
            background-color: ${props => props.theme.colors.neutral.white};
            color: ${props => props.theme.colors.primary};
        `}
    ${props =>
        props.disabled &&
        css`
            cursor: not-allowed;
            color: ${props.theme.colors.neutral.medium};
        `}
    ${props =>
        props.error &&
        css`
            color: ${props => props.theme.colors.suport.error};
        `}
`

export const Input = styled.div<InputComponentStylesProps>`
    display: flex;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    padding-left: 0.8rem;
    padding-right: 0.8rem;

    ${props =>
        props.inputSize === 'large' &&
        css`
            border: 2px solid ${props.theme.colors.secondary};
        `}

    ${props =>
        props.inputSize === 'medium' && props.isFilled
            ? css`
                  border: 2px solid ${props.theme.colors.neutral.dark};
              `
            : css`
                  border: 2px solid ${props.theme.colors.neutral.soft};
              `}
    border-radius: ${props => props.theme.borderRadius};
    color: ${props => props.theme.colors.neutral.dark};
    align-items: center;
    justify-content: center;
    input {
        ${props => props.theme.fontTokens[100]}
        font-family: inherit;
        width: 100%;
        height: 88%;
        outline: none;
        background: none;
        border-style: none;
        &::-ms-reveal,
        &::-ms-clear {
            display: none;
        }
        ${props =>
            props.isActive &&
            props.error &&
            css`
                color: ${props => props.theme.colors.suport.error};
            `}
        ${props =>
            props.disabled &&
            css`
                cursor: not-allowed;
            `}
    }
    ${props =>
        props.error &&
        css`
            border-color: ${props => props.theme.colors.suport.error};
        `}
    ${props =>
        props.isActive &&
        props.error &&
        css`
            border-color: ${props => props.theme.colors.suport.error};
            color: ${props => props.theme.colors.suport.error};
        `}
    ${props =>
        props.isActive &&
        !props.error &&
        css`
            border-color: ${props => props.theme.colors.primary};
        `}
    ${props =>
        props.disabled &&
        css`
            cursor: not-allowed;
            background-color: ${props.theme.colors.neutral.light[300]};
            border-color: ${props.theme.colors.neutral.light[300]};
        `}
`


export const TopContainer = styled.div<InputComponentStylesProps>`
    box-sizing: border-box;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    ${props => {

        if(props.inputSize === 'medium' && props.error)
            return css`height:74px;`
        else if(props.inputSize === 'medium' && !!props.label)
            return css`height:50px;`
        else if(props.inputSize === 'medium')
            return css`height:32px;`
        if(props.inputSize === 'large' && props.error)
            return css `height:85px;`
        return css`height:60px;`
    }};

    /* marginTop
    marginBottom,
    marginLeft,
    marginRight,
    width,
    height:
    inputSize === 'medium' &&
         !!error?.message &&
                    error?.message !== ''
                        ? '74px'
                        : '50px',
                display: 'flex',
                flexDirection: 'column',
                justifyContent: 'space-between' */
`
