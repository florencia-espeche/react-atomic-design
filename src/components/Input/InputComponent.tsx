import React, { useCallback, useEffect, useRef, useState } from 'react'
import {
    Container,
    AlertMessage,
    Input,
    Label,
    Svg,
    TopContainer,
    Mediumlabel
} from './InputComponent.styles'
import { InputProps } from './InputComponent.types'
import { Eye } from '@svg/Eye'
import { EyeOff } from '@svg/EyeOff'
import { AlertCircle } from '@svg/AlertCircle'
import { useTheme } from '../../hooks/useTheme'

const InputComponent = ({
    variant = 'text',
    label,
    error,
    type = 'text',
    marginTop,
    marginBottom,
    marginLeft,
    marginRight,
    width,
    inputSize = 'large',
    disabled = false,
    mask,
    value,
    ...rest
}: InputProps) => {
    const [isActive, setIsActive] = useState(false)
    const [isEyeOn, setIsEyeOn] = useState(true)
    const inputRef = useRef<HTMLInputElement>(null)
    const [filled, setFilled] = useState(false)

    const theme = useTheme()

    useEffect(() => {
        if (
            inputSize === 'large' &&
            inputRef.current?.value &&
            inputRef.current.value != ''
        ) {
            setIsActive(true)
        }
    })

    useEffect(() => {
        if (!!error && error.message) {
            setIsActive(true)
        }
    }, [error, inputRef])

    const heandleFocus = () => {
        setIsActive(true)
        inputRef.current?.focus()
    }

    const handleBlur = () => {
        if (inputSize === 'medium' || inputRef.current?.value.length === 0) {
            setIsActive(false)
        }
        if (inputRef.current?.value.length > 0) {
            setFilled(true)
        } else {
            setFilled(false)
        }
    }

    const handleEyeClick = () => {
        setIsEyeOn(!isEyeOn)
        if (isEyeOn) {
            inputRef.current?.setAttribute('type', 'text')
        } else {
            inputRef.current?.setAttribute('type', 'password')
        }
    }

    const dateMask = (
        value?: string | ReadonlyArray<string> | number | undefined
    ) => {
        if (!value) return value

        let newValue = `${value}`

        newValue = newValue.replace(/\D/g, '')

        if (newValue.length <= 3) newValue = newValue.replace(/(\d{2})/, '$1/')

        if (newValue.length === 4)
            newValue = newValue.replace(/(\d{2})(\d{2})/, '$1/$2/')

        if (newValue.length > 4)
            newValue = newValue.replace(/(\d{2})(\d{2})(\d)/, '$1/$2/$3')

        return newValue.substring(0, 10)
    }

    const timeMask = (
        value?: string | ReadonlyArray<string> | number | undefined
    ) => {
        if (!value) return value

        let newValue = `${value}`

        newValue = newValue.replace(/\D/g, '')

        if (newValue.length <= 3) newValue = newValue.replace(/(\d{2})/, '$1:')

        if (newValue.length === 4)
            newValue = newValue.replace(/(\d{2})(\d{2})/, '$1:$2:')

        if (newValue.length > 4)
            newValue = newValue.replace(/(\d{2})(\d{2})(\d)/, '$1:$2:$3')

        return newValue.substring(0, 8)
    }

    const handleMask = useCallback(
        (value: any) => {
            if (mask === 'date') {
                return dateMask(value)
            }

            if (mask === 'time') {
                return timeMask(value)
            }

            return value
        },
        [mask]
    )

    return (
        <TopContainer
            data-testid="input-top-container"
            variant={variant}
            label={label}
            inputSize={inputSize}
            error={!!(error && error?.message && error?.message !== '')}
            style={{
                marginTop,
                marginBottom,
                marginLeft,
                marginRight,
                width
            }}
        >
            <Container
                variant={variant}
                onClick={heandleFocus}
                onBlur={handleBlur}
                data-testid="input-container"
                theme={theme}
                disabled={disabled}
                inputSize={inputSize}
            >
                {inputSize === 'medium' && label && (
                    <Mediumlabel
                        isActive={isActive}
                        error={!!(error && error.message)}
                        data-testid="input-label"
                        theme={theme}
                        disabled={disabled}
                    >
                        {label}
                    </Mediumlabel>
                )}
                <Input
                    isFilled={filled}
                    isActive={isActive}
                    isEyeOn={isEyeOn && variant === 'password'}
                    error={!!(error && error.message)}
                    inputSize={inputSize}
                    theme={theme}
                    disabled={disabled}
                >
                    <input
                        data-testid="input"
                        ref={inputRef}
                        disabled={disabled}
                        type={variant === 'password' ? 'password' : 'text'}
                        value={handleMask(value)}
                        {...rest}
                    />
                    {variant === 'password' && (
                        <Svg
                            theme={theme}
                            onClick={() => handleEyeClick()}
                            data-testid="input-eye"
                        >
                            {isEyeOn ? Eye : EyeOff}
                        </Svg>
                    )}
                </Input>
                {inputSize === 'large' && (
                    <Label
                        theme={theme}
                        isActive={isActive}
                        error={!!(error && error.message)}
                        data-testid="input-label"
                        disabled={disabled}
                    >
                        {label}
                    </Label>
                )}
            </Container>
            {error && error.message && (
                <AlertMessage theme={theme} data-testid="input-errorMessage">
                    {AlertCircle}
                    {error?.message}
                </AlertMessage>
            )}
        </TopContainer>
    )
}

export default InputComponent
