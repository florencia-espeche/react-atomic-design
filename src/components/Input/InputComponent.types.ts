export interface InputProps
    extends React.InputHTMLAttributes<HTMLInputElement> {
    label?: string
    error?: InputError
    variant?: 'password' | 'text'
    marginTop?: string
    marginBottom?: string
    marginLeft?: string
    marginRight?: string
    inputSize?: 'large' | 'medium'
    disabled?: boolean
    mask?: 'date' | 'time'
}

export type InputError = {
    message?: string
    icon?: string
}
