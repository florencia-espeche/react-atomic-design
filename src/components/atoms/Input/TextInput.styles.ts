import styled, { css } from 'styled-components'

interface StyledWrapperProps {
    disabled?: boolean;
    inputSize: string;
    width: string;
}

interface StyledTextInputProps {
    error: boolean;
    width: string;
}

export const StyledWrapper = styled.div<StyledWrapperProps>`
    position: relative;
    width: ${pr => pr.width};
    ${props =>
        props.inputSize === 'large' &&
        css`
            height: 48px;
        `}
    ${props =>
        props.inputSize === 'medium' &&
        css`
            height: 32px;
        `}
    ${props =>
        props.disabled &&
        css`
            cursor: not-allowed;
        `}
`;

export const StyledTextInput = styled.input<StyledTextInputProps>`
    box-sizing: border-box;
    position: relative;
    background-color: ${ (pr) => pr.error ? '#ffe3e6' : '#EEEEEE' };
    padding: 0;
    width: ${pr => pr.width};
    border: none;
    color: #000;
    ${ pr => pr.disabled ? `
    background-color: #a6a6a6;
    color: #5e5e5e;
    cursor: not-allowed;
    
    &:hover {
        background-color: #a6a6a6 !important;
        color: #5e5e5e !important;
    }
    ` : ''}
    border-radius: 0;
    outline: none;
    transition: 0.1s ease-out;
    
    box-shadow: inset 0 0 0 2px ${pr => pr.error ? '#d93848' : 'transparent'};
    &:focus {
        box-shadow: inset 0 0 0 2px ${pr => pr.error ? '#d93848' : '#000'};
    }
`;
