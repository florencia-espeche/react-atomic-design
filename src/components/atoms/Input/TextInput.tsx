import React, { ChangeEventHandler, HTMLAttributes } from 'react'
import { StyledTextInput, StyledWrapper } from './TextInput.styles'
import { useTheme } from '../../../hooks/useTheme'

export interface TextInputProps
    extends HTMLAttributes<HTMLInputElement> {
    disabled?: boolean
    error?: boolean
    inputSize?: 'large' | 'medium' | 'small'
    value?: string
    width?: string
    onChange?: ChangeEventHandler<HTMLInputElement>
}

export const TextInput: React.ForwardRefRenderFunction<
    HTMLInputElement,
    TextInputProps
> = props => {
    const {
        disabled = false,
        error = false,
        inputSize = 'medium',
        value,
        onChange,
        placeholder,
        width = '100%',
        ...rest
    } = props

    const theme = useTheme()

    const styles = {
        disabled: disabled,
        error,
        width
    }

    return (
        <StyledWrapper inputSize={inputSize} width={width} theme={theme}>
            <StyledTextInput
                data-testid="input"
                value={value}
                onChange={onChange}
                placeholder={placeholder}
                theme={theme}
                {...styles}
                {...rest}
            />
        </StyledWrapper>
    )
}
