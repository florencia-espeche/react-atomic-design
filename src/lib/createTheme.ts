import { defaultTheme } from '../styles/defaultTheme'
import { Theme } from '../styles/types'

export const createTheme = <T>(theme: Theme & T) => {
    return {
        ...defaultTheme,
        colors: {
            ...defaultTheme.colors,
            ...theme.colors
        },
        borderRadius: theme.borderRadius || defaultTheme.borderRadius,
        fontTokens: {
            ...defaultTheme.fontTokens,
            ...theme.fontTokens
        },
        fontWeight: theme.fontWeight || defaultTheme.fontWeight,
        name: theme.name || defaultTheme.name
    }
}
