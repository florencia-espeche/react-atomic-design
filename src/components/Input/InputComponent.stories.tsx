import React from 'react'
import InputComponent from './InputComponent'

export default {
    title: 'Input',
    component: InputComponent,
    decorators: [
        Story => (
            <div style={{ width: '200px' }}>
                <Story />
            </div>
        )
    ],
    argTypes: {
        variant: {
            options: ['text', 'password'],
            control: {
                type: 'radio'
            }
        },
        value: {
            control: {
                type: 'text'
            }
        },
        label: {
            control: {
                type: 'text'
            }
        },
        error: {
            control: {
                type: 'object'
            }
        }
    },
    arg: {
        variant: 'text',
        label: 'Label'
    }
}

export const Default = args => {
    return (
        <div>
            <InputComponent {...args} />
            <InputComponent {...args} />
        </div>
    )
}
