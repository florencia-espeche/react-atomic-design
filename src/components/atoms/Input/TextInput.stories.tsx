import React, { useState } from 'react';
// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Story, Meta } from '@storybook/react/types-6-0';

import { TextInput, TextInputProps } from './TextInput';
import styled from 'styled-components';


export default {
  title: 'Atoms/TextInput',
  component: TextInput,
} as Meta;

const Template: Story<TextInputProps> = (args) => <TextInput {...args} />;

export const Default = Template.bind({});
Default.args = {
  width: '250px'
}

export const Placeholder = Template.bind({});
Placeholder.args = {
  placeholder: 'Placeholder',
  width: '250px'
}

export const WithError = Template.bind({});
WithError.args = {
  placeholder: 'Wrong input',
  width: '250px',
  error: true
}

export const Disabled = Template.bind({});
Disabled.args = {
  placeholder: 'Disabled',
  width: '250px',
  disabled: true,
}

const Row = styled.div`
  display: flex;
  align-items: flex-start;
  margin-bottom: 10px;
  & > * {
    margin-right: 10px;
  }
`;

export const Sizes = () => {
  return (
    <>
      <Row>
        <TextInput placeholder='large' inputSize='large' />
        <TextInput placeholder='medium' inputSize='medium' />
        <TextInput placeholder='small' inputSize='small' />
      </Row>
    </>
  );
};