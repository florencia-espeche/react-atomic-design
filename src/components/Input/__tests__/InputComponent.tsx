import React from 'react'
import 'jest-styled-components'

import Input from '../InputComponent'
import { fireEvent, render } from '../../../lib/test-utils'

jest.mock('@svg/AlertCircle', () => {
    return () => <></>
})

describe('Input Component', () => {
    it('Should be able to match a snapshot with false parameter', () => {
        const { container } = render(
            <Input
                label="Test"
                error={{
                    message: 'Test'
                }}
            />
        )

        expect(container.firstChild).toMatchSnapshot()
    })

    it('Should render the component with medium size', () => {
        const { container } = render(
            <Input
                label="Test"
                inputSize="medium"
                error={{
                    message: 'Test'
                }}
                variant="text"
            />
        )

        expect(container.firstChild).toMatchSnapshot()
    })

    it('Should be able to match a snapshot with true parameter', () => {
        const { container } = render(
            <Input
                label="Test"
                error={{
                    message: 'Test'
                }}
                variant="password"
            />
        )

        expect(container.firstChild).toMatchSnapshot()
    })

    it('Should be able to check the error message', () => {
        const errorMessage = 'Error Message'
        const { getByTestId } = render(
            <Input
                error={{
                    message: errorMessage
                }}
                label="Label"
                variant="password"
            />
        )
        const alertMessage = getByTestId('input-errorMessage')
        expect(alertMessage.innerHTML).toBe(errorMessage)
    })

    it('Should be not able to check the error message when message is diferent', () => {
        const errorMessage = 'Error Message'
        const { getByTestId } = render(
            <Input
                error={{
                    message: errorMessage
                }}
                label="Label"
                variant="password"
            />
        )
        const alertMessage = getByTestId('input-errorMessage')
        expect(alertMessage.innerHTML === 'Mensagem Diferente').toBeFalsy()
    })

    it('Should be able to change a value of input', () => {
        const { getByTestId } = render(
            <Input
                error={{
                    message: ''
                }}
                label="Label"
                variant="password"
                onChange={jest.fn}
            />
        )
        const input = getByTestId('input')
        const value = 'value'
        fireEvent.change(input, { target: { value } })
        expect((input as HTMLInputElement).value).toBe(value)
    })

    it('Should be able to click on eye', () => {
        const { getByTestId } = render(
            <Input
                error={{
                    message: ''
                }}
                label="Label"
                variant="password"
                onChange={jest.fn}
            />
        )
        const eye = getByTestId('input-eye')
        const input = getByTestId('input')

        fireEvent.click(eye)
        expect((input as HTMLInputElement).type).toBe('text')
        fireEvent.click(eye)
        expect((input as HTMLInputElement).type).toBe('password')
    })

    it('Should be able to click outside the input and trigger blur', () => {
        const { getByTestId } = render(
            <Input
                error={{
                    message: ''
                }}
                label="Label"
                variant="password"
            />
        )
        const input = getByTestId('input')
        input.focus()
        expect(input).toHaveFocus()
        input.blur()
        const label = getByTestId('input-label')
        expect(label).toMatchSnapshot()
    })

    it('Should render an input with date mask', () => {
        const { getByTestId } = render(
            <Input
                error={{
                    message: ''
                }}
                value="11102012"
                label="Label"
                mask="date"
                onChange={jest.fn}
            />
        )

        const input = getByTestId('input')

        const newInputValue = '11/10/2012'

        expect(input).toHaveValue(newInputValue)
    })

    it('Should render an input with time mask', () => {
        const { getByTestId } = render(
            <Input
                error={{
                    message: ''
                }}
                value="000000"
                label="Label"
                mask="time"
                onChange={jest.fn}
            />
        )

        const input = getByTestId('input')

        const newInputValue = '00:00:00'

        expect(input).toHaveValue(newInputValue)
    })
})
