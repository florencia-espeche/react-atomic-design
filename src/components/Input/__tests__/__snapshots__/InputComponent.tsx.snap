// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component Should be able to click outside the input and trigger blur 1`] = `
.c0 {
  position: absolute;
  left: 1rem;
  top: 35%;
  padding: 0 0.5rem;
  margin-left: -0.5rem;
  color: #A7ADBF;
  cursor: text;
  -webkit-transition: top 200ms ease-in,left 200ms ease-in,font-size 200ms ease-in;
  transition: top 200ms ease-in,left 200ms ease-in,font-size 200ms ease-in;
}

<label
  class="c0"
  data-testid="input-label"
>
  Label
</label>
`;

exports[`Input Component Should be able to match a snapshot with false parameter 1`] = `
.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding: 5px;
  color: #E73A33;
}

.c4 svg {
  margin-right: 5px;
}

.c1 {
  box-sizing: border-box;
  position: relative;
  width: auto;
  height: 48px;
}

.c3 {
  position: absolute;
  left: 1rem;
  top: 35%;
  padding: 0 0.5rem;
  margin-left: -0.5rem;
  color: #A7ADBF;
  cursor: text;
  -webkit-transition: top 200ms ease-in,left 200ms ease-in,font-size 200ms ease-in;
  transition: top 200ms ease-in,left 200ms ease-in,font-size 200ms ease-in;
  top: -0.5rem;
  font-size: 0.8rem;
  left: 0.8rem;
  background-color: #FFF;
  color: #2B52DD;
  color: #E73A33;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  padding-left: 0.8rem;
  padding-right: 0.8rem;
  border: 2px solid #004282;
  border: 2px solid #A7ADBF;
  border-radius: 4px;
  color: #2B2C32;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-color: #E73A33;
  border-color: #E73A33;
  color: #E73A33;
}

.c2 input {
  font-size: 14px;
  line-height: 17px;
  font-family: inherit;
  width: 100%;
  height: 88%;
  outline: none;
  background: none;
  border-style: none;
  color: #E73A33;
}

.c2 input::-ms-reveal,
.c2 input::-ms-clear {
  display: none;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 85px;
}

<div
  class="c0"
  data-testid="input-top-container"
  label="Test"
>
  <div
    class="c1"
    data-testid="input-container"
  >
    <div
      class="c2"
    >
      <input
        data-testid="input"
        type="text"
        value=""
      />
    </div>
    <label
      class="c3"
      data-testid="input-label"
    >
      Test
    </label>
  </div>
  <div
    class="c4"
    data-testid="input-errorMessage"
  >
    Test
  </div>
</div>
`;

exports[`Input Component Should be able to match a snapshot with true parameter 1`] = `
.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding: 5px;
  color: #E73A33;
}

.c5 svg {
  margin-right: 5px;
}

.c1 {
  box-sizing: border-box;
  position: relative;
  width: auto;
  height: 48px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 svg {
  right: -2.9rem;
  top: 0.9rem;
  cursor: pointer;
}

.c4 {
  position: absolute;
  left: 1rem;
  top: 35%;
  padding: 0 0.5rem;
  margin-left: -0.5rem;
  color: #A7ADBF;
  cursor: text;
  -webkit-transition: top 200ms ease-in,left 200ms ease-in,font-size 200ms ease-in;
  transition: top 200ms ease-in,left 200ms ease-in,font-size 200ms ease-in;
  top: -0.5rem;
  font-size: 0.8rem;
  left: 0.8rem;
  background-color: #FFF;
  color: #2B52DD;
  color: #E73A33;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  padding-left: 0.8rem;
  padding-right: 0.8rem;
  border: 2px solid #004282;
  border: 2px solid #A7ADBF;
  border-radius: 4px;
  color: #2B2C32;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-color: #E73A33;
  border-color: #E73A33;
  color: #E73A33;
}

.c2 input {
  font-size: 14px;
  line-height: 17px;
  font-family: inherit;
  width: 100%;
  height: 88%;
  outline: none;
  background: none;
  border-style: none;
  color: #E73A33;
}

.c2 input::-ms-reveal,
.c2 input::-ms-clear {
  display: none;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 85px;
}

<div
  class="c0"
  data-testid="input-top-container"
  label="Test"
>
  <div
    class="c1"
    data-testid="input-container"
  >
    <div
      class="c2"
    >
      <input
        data-testid="input"
        type="password"
        value=""
      />
      <div
        class="c3"
        data-testid="input-eye"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M1 12C1 12 5 4 12 4C19 4 23 12 23 12C23 12 19 20 12 20C5 20 1 12 1 12Z"
            fill-rule="evenodd"
            stroke="#6D7384"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            clip-rule="evenodd"
            d="M12 15C13.6569 15 15 13.6569 15 12C15 10.3431 13.6569 9 12 9C10.3431 9 9 10.3431 9 12C9 13.6569 10.3431 15 12 15Z"
            fill-rule="evenodd"
            stroke="#6D7384"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </div>
    </div>
    <label
      class="c4"
      data-testid="input-label"
    >
      Test
    </label>
  </div>
  <div
    class="c5"
    data-testid="input-errorMessage"
  >
    Test
  </div>
</div>
`;

exports[`Input Component Should render the component with medium size 1`] = `
.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding: 5px;
  color: #E73A33;
}

.c4 svg {
  margin-right: 5px;
}

.c1 {
  box-sizing: border-box;
  position: relative;
  width: auto;
  height: 32px;
}

.c2 {
  color: #2B2C32;
  cursor: text;
  font-size: 12px;
  line-height: 15px;
  background-color: #FFF;
  color: #2B52DD;
  color: #E73A33;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  padding-left: 0.8rem;
  padding-right: 0.8rem;
  border: 2px solid #A7ADBF;
  border-radius: 4px;
  color: #2B2C32;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-color: #E73A33;
  border-color: #E73A33;
  color: #E73A33;
}

.c3 input {
  font-size: 14px;
  line-height: 17px;
  font-family: inherit;
  width: 100%;
  height: 88%;
  outline: none;
  background: none;
  border-style: none;
  color: #E73A33;
}

.c3 input::-ms-reveal,
.c3 input::-ms-clear {
  display: none;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 74px;
}

<div
  class="c0"
  data-testid="input-top-container"
  label="Test"
>
  <div
    class="c1"
    data-testid="input-container"
  >
    <div
      class="c2"
      data-testid="input-label"
    >
      Test
    </div>
    <div
      class="c3"
    >
      <input
        data-testid="input"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    class="c4"
    data-testid="input-errorMessage"
  >
    Test
  </div>
</div>
`;
